src/fireedge/src/modules/components/Forms/Datastore/CreateForm/Steps/ConfigurationAttributes/Fields/common.js

```diff
@@ -0,0 +1,185 @@
+  label: T.SafeDirs,
+  tooltip: [T.PressKeysToAddAValue, ['ENTER']],
+  type: INPUT_TYPES.AUTOCOMPLETE,
+  multiple: true,
+  validation: array(string().trim()).default(() => []),
+  dependOf: '$general.STORAGE_BACKEND',
+  htmlType: (type) =>
+    !typeIsOneOf(type, [isShared, isSsh, isCeph, isLvm]) && INPUT_TYPES.HIDDEN,
+  fieldProps: {
+    freeSolo: true,
+  },
+  grid: { xs: 12, md: 12 },
+}
+
+const STAGING_DIR = {
+  name: 'STAGING_DIR',
+  label: T.StagingDirectoryForImageRegistration,
+  type: INPUT_TYPES.TEXT,
+  dependOf: '$general.STORAGE_BACKEND',
+  htmlType: (type) =>
+    !typeIsOneOf(type, [isShared, isSsh, isCeph]) && INPUT_TYPES.HIDDEN,
+  validation: string().trim().notRequired(),
+  grid: { xs: 12, md: 12 },
+}
+
+/** @type {Field} - Storage limit field */
+const LIMIT_MB = {
+  name: 'LIMIT_MB',
+  label: T.StorageUsageLimit,
+  type: INPUT_TYPES.TEXT,
+  validation: number(),
+  dependOf: '$general.STORAGE_BACKEND',
+  htmlType: (type) =>
+    typeIsOneOf(type, [isShared, isSsh, isCeph, isLvm])
+      ? 'number'
+      : INPUT_TYPES.HIDDEN,
+  grid: { xs: 12, md: 6 },
+}
+
+/** @type {Field} - Transfer limit field */
+const LIMIT_TRANSFER_BW = {
+  name: 'LIMIT_TRANSFER_BW',
+  label: T.TransferBandwidthLimit,
+  type: INPUT_TYPES.TEXT,
+  validation: number(),
+  dependOf: '$general.STORAGE_BACKEND',
+  htmlType: (type) =>
+    typeIsOneOf(type, [isShared, isSsh, isCeph, isLvm])
+      ? 'number'
+      : INPUT_TYPES.HIDDEN,
+  grid: { xs: 12, md: 6 },
+}
+
+/** @type {Field} - LVM_THIN_ENABLE field */
+const LVM_THIN_ENABLE = {
+  name: 'LVM_THIN_ENABLE',
+  label: T.LvmThin,
+  type: INPUT_TYPES.SWITCH,
+  validation: boolean().yesOrNo(),
+  dependOf: '$general.STORAGE_BACKEND',
+  htmlType: (type) => !typeIsOneOf(type, [isLvm]) && INPUT_TYPES.HIDDEN,
+  grid: { xs: 12, md: 12 },
+}
+
+/** @type {Field} - No decompress field */
+const NO_DECOMPRESS = {
+  name: 'NO_DECOMPRESS',
+  label: T.DoNotTryToUntarOrDecompress,
+  type: INPUT_TYPES.SWITCH,
+  validation: boolean().yesOrNo(),
+  dependOf: '$general.STORAGE_BACKEND',
+  htmlType: (type) =>
+    !typeIsOneOf(type, [isShared, isSsh, isCeph, isLvm]) && INPUT_TYPES.HIDDEN,
+  grid: { xs: 12, md: 6 },
+}
+
+/** @type {Field} - Check capacity field */
+const DATASTORE_CAPACITY_CHECK = {
+  name: 'DATASTORE_CAPACITY_CHECK',
+  label: T.CheckDSCapacityBeforeCreatingImage,
+  type: INPUT_TYPES.SWITCH,
+  validation: boolean().yesOrNo(),
+  dependOf: '$general.STORAGE_BACKEND',
+  htmlType: (type) =>
+    !typeIsOneOf(type, [isShared, isSsh, isCeph, isLvm]) && INPUT_TYPES.HIDDEN,
+  grid: { xs: 12, md: 6 },
+}
+
+/** @type {Field} - Host bridge field */
+const BRIDGE_LIST = {
+  name: 'BRIDGE_LIST',
+  label: T.HostBridgeList,
+  tooltip: [T.PressKeysToAddAValue, ['ENTER']],
+  type: INPUT_TYPES.AUTOCOMPLETE,
+  multiple: true,
+  validation: array(string().trim())
+    .compact()
+    .default(() => [])
+    .when('$general.STORAGE_BACKEND', (storageBackend, schema) =>
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).